source/slang/syntax.h

```diff
@@ -293,6 +293,9 @@ namespace Slang
         RefPtr<Type>	type;
         bool					IsLeftValue;
 
+        template <typename T>
+        T* As();
```

Comment:
Ah, so maybe I'm missing something and those earlier cases were all using `QualType` instead of a `RefPtr<Type>`.